### DISCUSSION TITLE
component resets attribute values on updating traits ( first time only )

### DISCUSSION THREAD

**Discussion by rajevar** [GrapesJS/grapesjs#4759] [Q&A] [answered]
Hello @artf ,

When a component's traits are updated its class attribute resets to the default values. This only happens during the first time.  Please follow the steps in the jsfiddle to reproduce the issue.

https://jsfiddle.net/rajevar/zLs8q97f/4/

1. From the `blocks`, pick `Text CAT:Basic` component and drag to canvas.
2. HTML code after step-1 `  <div class="text-primary" id="ixci">Insert your text here  </div>`
3. Update class attribute for the text component to add a new class. `class-a`
4. HTML code after step-3 `<div class="text-primary class-a" id="ixci">Insert your text here </div>`
5. Go to component settings and change id to `text-id-1`
6. HTML code after step-5 `  <div id="text-id-1" class="text-primary">Insert your text here  </div>`
7. In the HTML, `class-a` got removed from `class` attribute. This happens for `title` change as well..
8. if the `id` or `title` changes again, the class values are preserved.
9. Repeat step-3 to add class `class-b`
10. Repeat step-5 to change `id` to 'text-id-2`
11. HTML code after step-10 `<div id="text-id-2" class="text-primary class-b">Insert your text here  </div>`

_Expected html code after step-5_ ` <div id="text-id-1" class="text-primary class-a">Insert your text here  </div>`

This issue only happens if the component has `class` attribute defined in the component definition. 
Appreciate any help in resolving this issue.

Thank you in advance.

**Comment by artf** [comment, marked as answer]
Thanks, good catch, I found the issue and it will be fixed in the next release.
